job-application-tracker-api/src/users/users.service.ts

```diff
@@ -50,24 +49,24 @@ export class UsersService {
         return UserMapper.toUserDto(savedUser);
     }
 
-    async update(id: string, updateUserDto: UpdateUserDto) {
-        const user = await this.userRepository.findById(id);
+    async update(applicationId: string, updateUserDto: UpdateUserDto) {
+        const user = await this.userRepository.findById(applicationId);
         if (!user) {
             throw new NotFoundException('User not found');
         }
         user.userName = updateUserDto.userName;
         user.firstName = updateUserDto.firstName;
         user.lastName = updateUserDto.lastName;
         user.email = updateUserDto.email;
-        let emailUser=this.userRepository.findByEmail(user.email);
-        if(emailUser){
-            if((await emailUser)._id !== id)
+        const emailUser = this.userRepository.findByEmail(user.email);
+        if (emailUser) {
+            if ((await emailUser)._id !== applicationId)
                 throw new BadRequestException('this email is being used by a different user');
         }
         if (updateUserDto.password) {
             user.passwordHash = await hash(updateUserDto.password, 10);
         }
-        const updatedUser = await this.userRepository.update(id,user);
+        const updatedUser = await this.userRepository.update(applicationId, user);
```

Comment:
Why is this called applicationId

Reply:
באמת שאין לי מושג מה עשיתי פה

Reply:
נראלי פשוט התבלבלתי בסרבסים אבל אין לי שמץ